File: api/app/schemas/vehicles.py
from pydantic import BaseModel
from typing import List
from datetime import date

class VehicleMetrics(BaseModel):
    vehicle_id: str
    name: str
    total_trips: int
    total_mileage: float
    total_fuel: float
    fuel_efficiency: float
    shipments_delivered: int
    avg_delivery_time: float
    total_revenue: float
    revenue_per_trip: float
    fuel_per_trip: float

class RoutePerformance(BaseModel):
    origin: str
    destination: str
    route_trips: int
    avg_route_time: float
    avg_route_revenue: float

class DailyPerformance(BaseModel):
    date: date
    trips: int
    daily_mileage: float | None
    daily_fuel: float | None
    daily_revenue: float | None
    daily_deliveries: int

class VehicleDetailSummary(BaseModel):
    vehicle_id: str
    name: str
    lifetime_mileage: float
    total_trips: int
    period_mileage: float
    total_fuel: float
    fuel_efficiency: float
    deliveries_completed: int
    avg_delivery_time: float
    total_revenue: float
    revenue_per_trip: float
    fuel_per_trip: float

class VehicleDetailsResponse(BaseModel):
    summary: VehicleDetailSummary
    daily_performance: List[DailyPerformance]
    top_routes: List[RoutePerformance] 

File: client/src/types/vehicles.ts
export interface VehicleDetailsRequest {
  id: string;
  start: string;
  end: string;
}

export interface VehicleMetric {
  vehicle_id: string;
  name: string;
  total_trips: number;
  total_mileage: number;
  total_fuel: number;
  fuel_efficiency: number;
  shipments_delivered: number;
  avg_delivery_time: number;
  total_revenue: number;
  revenue_per_trip: number;
  fuel_per_trip: number;
}

export interface VehicleDetailSummary {
  vehicle_id: string;
  name: string;
  lifetime_mileage: number;
  total_trips: number;
  period_mileage: number;
  total_fuel: number;
  fuel_efficiency: number;
  deliveries_completed: number;
  avg_delivery_time: number;
  total_revenue: number;
  revenue_per_trip: number;
  fuel_per_trip: number;
}

export interface DailyPerformance {
  date: string;
  trips: number;
  daily_mileage: number;
  daily_fuel: number;
  daily_revenue: number;
  daily_deliveries: number;
}

export interface RoutePerformance {
  origin: string;
  destination: string;
  route_trips: number;
  avg_route_time: number;
  avg_route_revenue: number;
}

export interface VehicleDetail {
  summary: VehicleDetailSummary;
  daily_performance: DailyPerformance[];
  top_routes: RoutePerformance[];
}

export interface VehicleState {
  metrics: {
    data: VehicleMetric[] | null;
    loading: boolean;
    error: string | null;
  };
  detail: {
    data: VehicleDetail | null;
    loading: boolean;
    error: string | null;
  }
}